### PR TITLE
fix: klog verbosity detection

### LINKED ISSUE #1314: No log output
**What happened**:
add --v=X, X is log level. There is no log output for any level.

**What you expected to happen**:
option --v=X works well

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`)
- KubeFed version
master
- Scope of installation (namespaced or cluster)
- Others

/kind bug

### PR DESCRIPTION
**Pull request by hectorj2f**
Signed-off-by: Hector Fernandez <hfernandez@mesosphere.com>

**What this PR does / why we need it**:

The verbosity level was not initialized to klog anymore, so the level of verbosity was the default one instead of the specified in the flags.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1314 

**Special notes for your reviewer**:

### CODE REVIEW

File: cmd/controller-manager/app/controller-manager.go [hunk 1]
```diff
@@ -93,16 +93,18 @@ member clusters and do the necessary reconciliation`,
...
+	flags.StringVar(&healthzAddr, "healthz-addr", healthzDefaultBindAddress, "The address the healthz endpoint binds to.")
+	flags.StringVar(&metricsAddr, "metrics-addr", metricsDefaultBindAddress, "The address the metric endpoint binds to.")
+	flags.BoolVar(&verFlag, "version", false, "Prints the Version info of controller-manager.")
+	flags.StringVar(&kubeFedConfig, "kubefed-config", "", "Path to a KubeFedConfig yaml file. Test only.")
+	flags.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+	flags.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+
+	local := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	klog.InitFlags(local)
+	flags.AddGoFlagSet(local)
```
Comment (jimmidyson) [inline cmd/controller-manager/app/controller-manager.go:107]: The pattern I've seen adopted more in https://github.com/kubernetes/kubernetes repo is like:

```
# First line of main
klog.InitFlags(nil)
...
# Do other flag stuff
...
cmd.AddGoFlagSet(flag.CommandLine)
```

Comment (hectorj2f) [reply cmd/controller-manager/app/controller-manager.go:107]: The pattern shared in this PR comes from the Kubernetes repo. Check how the initialize klog flags.

Comment (hectorj2f) [reply cmd/controller-manager/app/controller-manager.go:107]: Could you detail what you see wrong in the current pattern?

**Comment by mYmNeo** [comment]
@hectorj2f I've tried, it works well. :+1:

File: cmd/controller-manager/app/controller-manager.go (hunk 1 above)
Comment (makkes) [reply cmd/controller-manager/app/controller-manager.go:107]: Stupid question: Is there a difference between first initializing klog flags and then others as opposed to doing it the other way around? If not, I'm fine either way.

**Review by makkes — APPROVED** [review]

File: cmd/controller-manager/app/controller-manager.go (hunk 1 above)
Comment (jimmidyson) [reply cmd/controller-manager/app/controller-manager.go:107]: Yeah either work, I was just pointing the more common approach (see https://github.com/kubernetes/kubernetes/search?q=%22klog.InitFlags%22) but it's not a biggie.

**Pull request merged**